grpc/grpc.go

```diff
@@ -80,3 +86,69 @@ func (mi *ModuleInstance) Exports() modules.Exports {
 		Named: mi.exports,
 	}
 }
+
+// stream returns a new stream object
+func (mi *ModuleInstance) stream(c goja.ConstructorCall) *goja.Object {
+	rt := mi.vu.Runtime()
+
+	// the first argument is the GRPC client
+	obj := c.Argument(0).ToObject(rt).Export()
+
+	client, ok := obj.(*Client)
+	if !ok {
+		common.Throw(rt, errors.New("the first argument must be a GRPC client"))
+	}
+
+	// the second argument is the method name
+	methodName := sanitizeMethodName(c.Argument(1).String())
+	methodDescriptor, err := client.getMethodDescriptor(methodName)
+	if err != nil {
+		common.Throw(rt, err)
+	}
+
+	// the third argument is the params (optional)
+	// TODO: abstract logic for parsing params
+	// should be something similar to the Invoke function
+	// in js/modules/k6/grpc/client.go
+
+	registerCallback := func() func(func() error) {
+		callback := mi.vu.RegisterCallback()
+		return func(f func() error) {
+			callback(f)
+		}
+	}
+
+	tagsAndMeta := mi.vu.State().Tags.GetCurrentValues()
+
+	s := &stream{
+		vu:               mi.vu,
+		client:           client,
+		methodDescriptor: methodDescriptor,
+		method:           methodName,
+		logger:           mi.vu.State().Logger,
+
+		tq: taskqueue.New(registerCallback),
+
+		builtinMetrics: mi.vu.State().BuiltinMetrics,
+
+		closed: false,
+		done:   make(chan struct{}),
+
+		writeQueueCh: make(chan message, 10),
```

Comment:
10 is a simple buffer. There are no plans to write the logging and configurations for now, but if you think that could be an issue, I could try to adjust the logic.

Reply:
I got it, but is it a benchmarked number or just a value to get some buffering? I think we should start with just 1 for the minimum async (or the number of senders if they differ from 1 and increment if we can't get a good rate during benchmarks).
Otherwise, in this way, we could hide if the writer(s) got stuck and not realize it.

Reply:
I believe it's the value to get some buffering. It's the same number we use in the xk6-websockets, for the same purposes https://github.com/grafana/xk6-websockets/blob/main/websockets/websockets.go#L123 I don't think we did a benchmark there.

To be honest, I see your concerns, but not sure that they are significant to make an adjustment in that particular PR. Are you fine with creating a separate task and investigating/addressing them there?

Reply:
I prefer to do the opposite as described in the previous comment, but I don't want to block you on this. I'm fine with your suggestion too.

Reply:
@mstoykov, what do you think? If you share Ivan's concerns, I could re-think that in that PR

Reply:
@codebien seems to be right. I don't remember why I have chosen a `10` here and there is an additional queue either way.

I doubt there is any real impact by having both or having only 1 of the bufferings. And this likely should be "fixed" in the websockets code as well

Reply:
:+1: 

Okay, I will add the go leak check here and adopt the latest webscoket's changes to the PR. So will fix it in following commits :+1: